File: floating-point-number/step-number.py
from struct import *

n = 0x3ff0000000000000

print("  double in binary => text representation")
for _ in range(1000):
    float_value = unpack('d', pack('Q', n))[0]
    print("  %016x => %s" % (n, float_value))
    n += 1

File: floating-point-number/step-number.rs
fn main() {
    let mut n: u64 = 0x3ff0000000000000;

    println!("  double in binary => text representation");
    for _ in 0..1000 {
        let float_value: f64 = unsafe { std::mem::transmute(n) };
        println!("  {:016x} => {}", n, float_value);
        n += 1;
    }
}
